src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6804,10 +6804,16 @@ private BoundExpression BindArrayAccess(ExpressionSyntax node, BoundExpression e
                 }
             }
 
-            return new BoundArrayAccess(node, expr, convertedArguments.AsImmutableOrNull(), arrayType.ElementType.TypeSymbol, hasErrors);
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            var resultType = rank == 1 &&
+                convertedArguments[0].Type == GetWellKnownType(WellKnownType.System_Range, ref useSiteDiagnostics)
```

Comment:
Given this isn't used can we use a discard here?

Reply:
Can't pass a discard by ref.

Reply:
Hmm ... we should fix that 

---
In reply to: [226708094](https://github.com/dotnet/roslyn/pull/30618#discussion_r226708094) [](ancestors = 226708094)